safe-global/safe-wallet-web
# feat: recovery dispatch notifications

## What it solves

This adds recovery notifications and in doing so [hides the proposal widget when a proposal is in progress](https://www.notion.so/safe-global/Show-Account-recovery-in-progress-immediately-57206a130c004fa58f5425c91e6605a0).

## How this PR fixes it

This builds upon the recovery event bus, adding notifications for Recoverer recovery transactions as we do for "standard" transactions and messages:

- Recovery proposal
- Recovery execution
- Skipping of expired recovery attempts

Each of the above have notifications as such throughout their lifecycle:

1. "Confirm the execution in your wallet..."
2. "Validating..."
3. "Successfully validated. Querying..."

If the transaction reverts or fails, the following is displayed accordingly:

1. "Reverted. Please check your gas settings."
2. "Failed."

With the implementation of the above, it is now possible to know when a proposal is "in progress", e.g. being submitted to or validated on-chain. As such, the dashboard widget for Recoverers that allows for starting recovery is hidden when a proposal is "in progress" (see recording below).

## How to test it

1. Propose a recovery and observe the new notifications, but also the "Start recovery" header hidden during the proposal lifecycle.
2. Execute a recovery and observe the new notifications.
3. Skip expired recovery attempts and observe the new notifications.

## Screenshots

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/81e11245-3d87-4882-b309-bfa88d90898a)

![recovery notifications](https://github.com/safe-global/safe-wallet-web/assets/20442784/81abd125-4871-4a25-b63b-68cf7d1d920f)

## Checklist
